src/components/structures/login/Login.js

```diff
@@ -125,7 +123,7 @@ module.exports = React.createClass({displayName: 'Login',
             // logins so let's skip that for now).
             loginLogic.chooseFlow(0);
         }, function(err) {
-            self._setErrorTextFromError(err);
+            self._setStateFromError(err);
```

Comment:
It'll only set it to `true` if the response is 401 or 403 (I'd like to be more specific, but there's the synapse bug). I think it's probably best to reset it to `false` on other errors as we don't know that the username/password is still the problem.